spec/abstractops.html

```diff
@@ -1098,14 +1098,25 @@ <h1>ISO 8601 grammar</h1>
       TimeFraction :
           Fraction
 
-      TimeZoneUTCOffset :
-          UTCOffset
-          UTCDesignator
+      UTCOffsetWithSubMinuteComponents[Extended] :
+          Sign Hour[+Padded] TimeSeparator[?Extended] MinuteSecond TimeSeparator[?Extended] MinuteSecond Fraction?
 
-      TimeZoneUTCOffsetName[Extended] :
+      UTCOffsetMinutePrecision :
           Sign Hour[+Padded]
-          Sign Hour[+Padded] TimeSeparator[?Extended] MinuteSecond
-          Sign Hour[+Padded] TimeSeparator[?Extended] MinuteSecond TimeSeparator[?Extended] MinuteSecond Fraction?
+          Sign Hour[+Padded] TimeSeparator[+Extended] MinuteSecond
+          Sign Hour[+Padded] TimeSeparator[~Extended] MinuteSecond
+
+      UTCOffsetSubMinutePrecision :
+          UTCOffsetMinutePrecision
+          UTCOffsetWithSubMinuteComponents[+Extended]
+          UTCOffsetWithSubMinuteComponents[~Extended]
```

Comment:
I prefer (and chose for the spec text in this PR) "_parameterize UTCOffsetWithSubMinuteComponents independent of UTCOffsetMinutePrecision_" because it pushes the brackets down to lower-level productions.

Personally, I find the bracketed notation to be one of the most confusing parts of TC39 spec-ese, and I can't be the only one who feels this way. Given that readers will likely start at top-level productions and work their way downwards, I think that it's helpful to many readers (including me!) to delay showing bracket syntax for as many steps down as possible.

So I like the solution that keeps both `UTCOffsetMinutePrecision` and `UTCOffsetSubMinutePrecision` bracketless.

Also, doing this makes the change in #2607 easier to review, because it's a simple swap of `UTCOffsetSubMinutePrecision` for `UTCOffsetMinutePrecision` in the definition of `TimeZoneUTCOffsetName`. 

Anyway, I'd like to leave this text as-is, and after the pressure of the next TC39 meeting is over, then we can refactor further as needed.

So I'll resolve this conversation for now. Feel free to reopen if you disagree!